ilpy/impl/solvers/ScipBackend.cpp

```diff
@@ -183,10 +183,12 @@ ScipBackend::addConstraint(const Constraint& constraint) {
 			lhs,
 			rhs));
 
-	_constraints.push_back(c);
 
 	SCIP_CALL_ABORT(SCIPaddCons(_scip, c));
-	SCIP_CALL_ABORT(SCIPreleaseCons(_scip, &c));
```

Comment:
If we don't release the constraint here (which is fine), we will have to make sure to do so in the destructor of `ScipBackend` (by calling `freeConstraints()`). Otherwise there will be a memory leak.

Alternatively, we can free them here. The `_scip` model will still keep its own reference and free the constraint (presumably) after `SCIPdelCons`.

Reply:
ah, of course! thanks for catching

Reply:
actually... isn't it already in the destructor?

https://github.com/funkelab/ilpy/blob/3385d846d74354ba1315c145a052c581fd963dda/ilpy/impl/solvers/ScipBackend.cpp#L22-L29

Reply:
It is! All good then.